pandas/core/indexes/numeric.py

```diff
@@ -144,15 +145,19 @@ def _ensure_array(cls, data, dtype, copy: bool):
                 data = list(data)
 
             orig = data
-            data = np.asarray(data, dtype=dtype)
+            if isinstance(data, (list, tuple)) and dtype is None:
+                if len(data):
+                    data = sanitize_array(data, index=None)
```

Comment:
`sanitize_array` is the same function we use in `Series.__new__` for list-like data, so makes sense to use here also.

Reply:
Yes, actually, fixed in newest push. Thanks.